Eigen/src/Core/util/Memory.h

```diff
@@ -223,12 +223,17 @@ EIGEN_DEVICE_FUNC inline void aligned_free(void *ptr)
   * \brief Reallocates an aligned block of memory.
   * \throws std::bad_alloc on allocation failure
   */
+EIGEN_DEVICE_FUNC
 inline void* aligned_realloc(void *ptr, std::size_t new_size, std::size_t old_size)
 {
   EIGEN_UNUSED_VARIABLE(old_size);
 
   void *result;
-#if (EIGEN_DEFAULT_ALIGN_BYTES==0) || EIGEN_MALLOC_ALREADY_ALIGNED
+#if defined(EIGEN_CUDA_ARCH)
+  result = malloc(new_size);
```

Comment:
@vkhristenko do have to do something to guarantee the alignment ?

Reply:
good question - no. on the device no. i don't know how this will be handled on the device. Meaning, i do not know if malloc on the card will give me precisely the requested amount/pad it. I tried to search but did not find a solution back then